.NET/Microsoft.Recognizers.Text.Number/Japanese/Parsers/JapaneseNumberParser.cs

```diff
@@ -0,0 +1,344 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.Recognizers.Text.Number.Japanese
+{
+    public class JapaneseNumberParser : BaseNumberParser
+    {
+        protected new readonly JapaneseNumberParserConfiguration Config;
+
+        public JapaneseNumberParser(JapaneseNumberParserConfiguration config) : base(config)
+        {
+            this.Config = config;
+        }
+
+        public override ParseResult Parse(ExtractResult extResultJpn)
+        {
+            string extra = null;
+            ParseResult ret = null;
+            extra = extResultJpn.Data as string;
+
+            var getExtResultJpn = new ExtractResult()
+            {
+                Start = extResultJpn.Start,
+                Length = extResultJpn.Length,
+                Data = extResultJpn.Data,
+                Text = extResultJpn.Text,
+                Type = extResultJpn.Type
+            };
+            
+            if (extra == null)
+            {
+                return null;
+            }
+
+            if (extra.Contains("Ordinal"))
+            {
+                ret = OrdParse(getExtResultJpn);
+            }
+            else if (extra.Contains("Per"))
+            {
+                ret = PerParse(getExtResultJpn);
+            }
+
+            if (ret != null)
+            {
+                ret.Text = extResultJpn.Text;
+            }
+
+            return ret;
+        }
+
+        // Replace full digtal numbers with half digtal numbers. "４" and "4" are both legal in Japanese, replace "４" with "4", then deal with "4"
```

Comment:
What does "full" and "half" mean?
Isn't there similar code in the Chinese version? Both should use the same method, unless there is something that's Japanese-specific.

Reply:
"full" and "half" mean half width and full width of character in Japanese, it's similar in Chinese

Reply:
Got it.
Then the method name must be more specific. Also, if this is the same code as for Chinese. It should be in a common place and reused from both.

Reply:
NormalizeCharWidth could be a reasonable name.

Reply:
OK, I'm working on public Class for Japanese and Chinese.